lib/matplotlib/transforms.py

```diff
@@ -2902,14 +2902,20 @@ def interval_contains(interval, val):
         A 2-length sequence, endpoints that define the interval.
     val : scalar
         Value to check is within interval.
+    rtol : scalar
+        Tolerance slippage allowed outside of this interval.  Default
+        0 * (b - a).
 
     Returns
     -------
     bool
-        Returns true if given val is within the interval.
+        Returns true if given val is within the interval (with tolerance)
     """
     a, b = interval
-    return a <= val <= b or a >= val >= b
+    if a > b:
+        a, b = b, a
+    rtol = (b - a) * rtol
```

Comment:
This breaks `interval_contains((inf, inf), inf)`, which used to return true, but is now false.

Perhaps a new `interval_contains_approx` function would make more sense

Reply:
Is that really something we need to support?  I'm not sure either True or False make sense here, and if anything I'd argue that should return False.

Reply:
Since `inf == inf` is true, `interval_contains((inf, inf), inf)` should also be true, also for the intervals `(inf, a)` and `(a, inf)`. Not sure if we really *need* that, but it's good to have this logically clean. Might prevent some bug in the future. No harm in adding a special-cased line for this.

Reply:
... the original function is now unchanged, and `interval_contains_close` has the tolerance-awareness now.